packages/web-frontend/src/containers/MeasureLegend/LegendEntry.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+const Key = styled.div`
+  padding: 5px;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  pointer-events: auto;
+  cursor: pointer;
+  ${p => (p.hidden ? 'opacity: 0.5;' : '')}
+`;
+
+const LegendEntry = ({ marker, label, value, dataKey, onClick, hiddenMeasures, unClickable }) => {
+  const hidden = (hiddenMeasures[dataKey] || {})[value];
+
+  return (
+    <Key onClick={unClickable ? null : () => onClick(dataKey, value, !hidden)} hidden={hidden}>
+      {marker}
+      <div>{label}</div>
+    </Key>
+  );
+};
+
+const mapLegendStateToProps = () => {
+  return state => ({
+    hiddenMeasures: state.map.measureInfo.hiddenMeasures,
+  });
+};
+
+const mapLegendDispatchToProps = () => {
+  return dispatch => ({
+    onClick: (key, value, hide) =>
+      dispatch({
+        key,
+        value,
+        type: hide ? 'HIDE_MAP_MEASURE' : 'UNHIDE_MAP_MEASURE',
+      }),
+  });
+};
+
+LegendEntry.propTypes = {
```

Comment:
I believe it is more conventional to put the propTypes before the mapState functions as they belong to the 'unconnected' component. It is not really consistently applied in web-frontend but if you look at `SearchBar/index.js` you will see a good example.

Reply:
That actually makes a lot of sense!